src/util/integerfactoring.h

```diff
@@ -0,0 +1,241 @@
+/* Include Guards */
+#ifndef INTEGERFACTORING_H
+#define INTEGERFACTORING_H
+
+
+/* Includes */
+#include <stdio.h>
+#include <stdint.h>
+#include <stdint-gcc.h>
+
+
+/* Defines */
+
+
+
+/* C++ Extern "C" Guard */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+
+/* Data Structure Prototypes & Typedefs */
+struct GA_FACTOR_LIST;
+typedef struct GA_FACTOR_LIST GA_FACTOR_LIST;
```

Comment:
@abergeron Incidentally, why must the typedef and struct not share the same name? Doing this is valid in C; After `typedef struct X X;`, using `X` is shorthand for `struct X`, but if you choose to do so you can still use `struct X` as a visual cue of its `struct`-ness.